WebDriverAgentLib/Utilities/FBConfiguration.m

```diff
@@ -411,4 +434,22 @@ + (BOOL)reduceMotionEnabled
   }
   return NO;
 }
+
+/**
+ Default parameters for traversing elements tree from parents to children while requesting XCElementSnapshot.
+ The original maxDepth parameter in a dictionary provided by XCAXClient_iOS's method defaultParams is set to INT_MAX.
+ This method returns parameters with maxDepth set to FBDefaultSnapshotRequestMaxDepth
+
+ @return dictionary with default parameters for element's snapshot request
+*/
++ (NSMutableDictionary *)defaultSnapshotRequestParameters
```

Comment:
I would say we need to make the returned dictionary static and init it once in `dipatch_once` section. Only `maxDepth` should be set outside of this section. The reason for this is the `snapshotRequestParameters`, where we retrieve the pointer to the result of this method only once, which means all the following changed of parameters to it won't affect the result of the swizzled method

Reply:
please could you clarify. 
currently : ```FBSnapshotRequestParameters``` is static and created only once in ```snapshotRequestParameters``` in ```dispatch_once_t``` section.
Swizzled method always returns ```FBSnapshotRequestParameters```.
Also, initially, there needs to be a default value for ```maxDepth``` param.

Reply:
is it - should it better to inline the method ```defaultSnapshotRequestParameters``` ? or i need coffee? :-)

Reply:
like so?
```
+ (NSDictionary *)snapshotRequestParameters
{
  static dispatch_once_t onceToken;
  dispatch_once(&onceToken, ^{
    FBSnapshotRequestParameters = [NSMutableDictionary dictionaryWithDictionary:@{
      @"maxArrayCount": @INT_MAX,
      @"maxChildren": @INT_MAX,
      @"maxDepth": @(FBSnapshotRequestMaxDepth),
      @"traverseFromParentsToChildren": @1
    }];
  });

  return FBSnapshotRequestParameters;
}
```

Reply:
Exactly.

Actually, the current approach would also work as expected, it is just less obvious that `defaultSnapshotRequestParameters` would only be called once. Perhaps, it makes sense to move the initialisation of the dictionary into the `load` callback.

We could also probably make the FBSnapshotRequestMaxDepth variable obsolete, since we anyway keep it in the FBSnapshotRequestParameters dictionary

Reply:
ye, was thinking about that. now looks better. please take a look